python_ta/cfg/visitor.py

```diff
@@ -58,3 +65,49 @@ def visit_if(self, node: astroid.If) -> None:
         self.cfg.link_or_merge(end_else, after_if_block)
 
         self._current_block = after_if_block
+
+    def visit_while(self, node: astroid.While) -> None:
+        old_curr = self._current_block
+
+        # Handle "test" block
+        test_block = self.cfg.create_block()
+        test_block.add_statement(node.test)
+        self.cfg.link_or_merge(old_curr, test_block)
+
+        after_while_block = self.cfg.create_block()
+
+        # add while scope
+        self._scope.append(('while', test_block, after_while_block))
+
+        # Handle "body" branch
+        body_block = self.cfg.create_block(test_block)
+        self._current_block = body_block
+        for child in node.body:
+            child.accept(self)
+        end_body = self._current_block
+        self.cfg.link_or_merge(end_body, test_block)
+
+        # step outside loop
+        self._scope.pop()
+
+        # Handle "else" branch
+        else_block = self.cfg.create_block(test_block)
+        self._current_block = else_block
+        for child in node.orelse:
+            child.accept(self)
+        end_else = self._current_block
+
+        self.cfg.link_or_merge(end_else, after_while_block)
+        self._current_block = after_while_block
+
+    def visit_break(self, node: astroid.Break) -> None:
+        old_curr = self._current_block
+        old_curr.add_statement(node)
+        loop_scope = None
+        for scope in self._scope:
```

Comment:
Set the loop in reverse order (you can do `reversed(self._scope)` here. This way you can `break` (or early `return`) the first time a loop node is encountered. You can even use an `else` attached to the loop!

Also you can use tuple unpacking right in the for loop: `for node, exits in ...`.